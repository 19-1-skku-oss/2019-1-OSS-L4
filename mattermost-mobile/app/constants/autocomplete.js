// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const AT_MENTION_REGEX = /\B(@([^@\r\n\s]*))$/i;

export const AT_MENTION_SEARCH_REGEX = /\bfrom:\s*(\S*)$/i;

export const CHANNEL_MENTION_REGEX = /\B(~([^~\r\n]*))$/i;

export const CHANNEL_MENTION_SEARCH_REGEX = /\b(?:in|channel):\s*(\S*)$/i;

export const DATE_MENTION_SEARCH_REGEX = /\b(?:on|before|after):\s*(\S*)$/i;

export const ALL_SEARCH_FLAGS_REGEX = /\b\w+:/g;