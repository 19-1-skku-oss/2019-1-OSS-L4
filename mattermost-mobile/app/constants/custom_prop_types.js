// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import PropTypes from 'prop-types';

export const Children = PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf([PropTypes.node])]);

export const Style = PropTypes.oneOfType([
    PropTypes.object, // inline style
    PropTypes.number, // style sheet entry
    PropTypes.array,
]);

export default {
    Children,
    Style,
};
