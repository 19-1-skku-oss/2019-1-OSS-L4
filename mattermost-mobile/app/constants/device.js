// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeviceInfo from 'react-native-device-info';
import RNFetchBlobFS from 'rn-fetch-blob/fs';
import keyMirror from 'mattermost-redux/utils/key_mirror';

const deviceTypes = keyMirror({
    CONNECTION_CHANGED: null,
    DEVICE_DIMENSIONS_CHANGED: null,
    DEVICE_TYPE_CHANGED: null,
    DEVICE_ORIENTATION_CHANGED: null,
    STATUSBAR_HEIGHT_CHANGED: null,
});

export default {
    ...deviceTypes,
    DOCUMENTS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Documents`,
    IMAGES_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Images`,
    IS_IPHONE_X: DeviceInfo.getModel().includes('iPhone X'),
    IS_TABLET: DeviceInfo.isTablet(),
    VIDEOS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Videos`,
};
