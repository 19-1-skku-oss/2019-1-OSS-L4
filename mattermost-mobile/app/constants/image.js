// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
