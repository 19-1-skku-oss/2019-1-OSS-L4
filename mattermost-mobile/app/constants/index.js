// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import PermissionTypes from './permissions';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    PermissionTypes,
    UpgradeTypes,
    ViewTypes,
};
