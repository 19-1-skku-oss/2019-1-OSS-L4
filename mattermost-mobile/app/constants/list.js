// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const VISIBILITY_CONFIG_DEFAULTS = {
    viewAreaCoveragePercentThreshold: 3,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
