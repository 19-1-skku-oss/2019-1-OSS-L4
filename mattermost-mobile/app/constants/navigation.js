// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
});

export default NavigationTypes;
