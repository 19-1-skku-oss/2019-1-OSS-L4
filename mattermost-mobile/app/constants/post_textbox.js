// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_CONTENT_HEIGHT = 100;
export const MAX_FILE_COUNT = 5;
export const IS_REACTION_REGEX = /(^\+:([^:\s]*):)$/i;
export const INSERT_TO_DRAFT = 'insert_to_draft';
export const INSERT_TO_COMMENT = 'insert_to_comment';
