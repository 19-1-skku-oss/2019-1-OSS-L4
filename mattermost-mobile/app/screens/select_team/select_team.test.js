// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import {RequestStatus} from 'mattermost-redux/constants';

import SelectTeam from './select_team.js';

jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

const getTeams = async () => {
    return {
        error: {},
    };
};

describe('SelectTeam', () => {
    const actions = {
        getTeams,
        handleTeamChange: jest.fn(),
        joinTeam: jest.fn(),
        logout: jest.fn(),
    };

    const baseProps = {
        actions,
        currentChannelId: 'someId',
        currentUrl: 'test',
        navigator: {
            setOnNavigatorEvent: jest.fn(),
        },
        userWithoutTeams: false,
        teams: [],
        theme: Preferences.THEMES.default,
        teamsRequest: {
            status: RequestStatus.FAILURE,
        },
    };

    test('should match snapshot for fail of teams', async () => {
        const wrapper = shallow(
            <SelectTeam {...baseProps}/>,
        );
        expect(wrapper.state('loading')).toEqual(true);
        await getTeams();
        expect(wrapper.state('loading')).toEqual(false);
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for teams', async () => {
        const props = {
            ...baseProps,
            teams: [{
                id: 'kemjcpu9bi877yegqjs18ndp4r',
                invite_id: 'ojsnudhqzbfzpk6e4n6ip1hwae',
                name: 'test',
            }],
            teamsRequest: {
                status: RequestStatus.SUCCESS,
            },
        };

        const wrapper = shallow(
            <SelectTeam {...props}/>,
        );
        expect(wrapper.state('page')).toEqual(0);
        await getTeams();
        expect(wrapper.state('page')).toEqual(1);
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
