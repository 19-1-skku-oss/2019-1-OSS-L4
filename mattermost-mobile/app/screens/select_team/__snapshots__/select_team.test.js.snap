// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTeam should match snapshot for fail of teams 1`] = `
<FailedNetworkAction
  errorDescription={
    Object {
      "defaultMessage": "Make sure you have an active connection and try again.",
      "id": "mobile.failed_network_action.shortDescription",
    }
  }
  errorTitle={
    Object {
      "defaultMessage": "Team Not Found",
      "id": "error.team_not_found.title",
    }
  }
  onRetry={[Function]}
  theme={
    Object {
      "awayIndicator": "#ffbc42",
      "buttonBg": "#166de0",
      "buttonColor": "#ffffff",
      "centerChannelBg": "#ffffff",
      "centerChannelColor": "#3d3c40",
      "codeTheme": "github",
      "dndIndicator": "#f74343",
      "errorTextColor": "#fd5960",
      "linkColor": "#2389d7",
      "mentionBj": "#ffffff",
      "mentionColor": "#145dbf",
      "mentionHighlightBg": "#ffe577",
      "mentionHighlightLink": "#166de0",
      "newMessageSeparator": "#ff8800",
      "onlineIndicator": "#06d6a0",
      "sidebarBg": "#145dbf",
      "sidebarHeaderBg": "#1153ab",
      "sidebarHeaderTextColor": "#ffffff",
      "sidebarText": "#ffffff",
      "sidebarTextActiveBorder": "#579eff",
      "sidebarTextActiveColor": "#ffffff",
      "sidebarTextHoverBg": "#4578bf",
      "sidebarUnreadText": "#ffffff",
      "type": "Mattermost",
    }
  }
/>
`;

exports[`SelectTeam should match snapshot for teams 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      <FormattedText
        defaultMessage="Open teams you can join"
        id="mobile.select_team.join_open"
        style={
          Object {
            "color": undefined,
            "fontSize": 13,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "height": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <CustomList
    data={Array []}
    extraData={false}
    listType="flat"
    loading={false}
    loadingComponent={
      <Loading
        color="grey"
        size="large"
        style={Object {}}
      />
    }
    onLoadMore={[Function]}
    onRefresh={[Function]}
    refreshing={false}
    renderItem={[Function]}
    shouldRenderSeparator={false}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;
